internal/storage/storage.go

```diff
@@ -0,0 +1,115 @@
+package storage
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+func ArchiveDir(dir string) (out string, err error) {
+	pwd, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	defer func() {
+		if deferErr := os.Chdir(pwd); deferErr != nil && err == nil {
+			err = deferErr
+		}
+	}()
+
+	if err := os.Chdir(filepath.Dir(dir)); err != nil {
+		return "", err
+	}
+
+	dir = "./"
+	if f, err := os.Stat(dir); os.IsNotExist(err) || !f.IsDir() {
+		return "nil", fmt.Errorf("invalid dir path: %s", dir)
+	}
+
+	tf, err := os.CreateTemp("", "tf-")
+	if err != nil {
+		return "", err
+	}
+	tmpName := tf.Name()
+	defer func() {
+		if err != nil {
+			os.Remove(tmpName)
+		}
+	}()
+
+	h := sha1.New()
+	mw := io.MultiWriter(h, tf)
+
+	gw := gzip.NewWriter(mw)
+	tw := tar.NewWriter(gw)
+	if err := filepath.Walk(dir, func(p string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Ignore anything that is not a file (directories, symlinks)
```

Comment:
I think symlinks can be used with Terraform, they can link modules, or just specific files between modules to make access "easier" to access and easier to maintain (edit only once).